Index: nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/AbstractRestResourceProvider.java
===================================================================
--- nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/AbstractRestResourceProvider.java	(revision 59996)
+++ nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/AbstractRestResourceProvider.java	(working copy)
@@ -53,8 +53,8 @@
 import org.glassfish.jersey.media.multipart.MultiPartFeature;
 import org.glassfish.jersey.message.MessageProperties;
 import org.glassfish.jersey.server.ResourceConfig;
+import org.glassfish.jersey.server.ServerProperties;
 import org.glassfish.jersey.server.filter.CsrfProtectionFilter;
-import org.glassfish.jersey.server.filter.UriConnegFilter;
 
 import javax.ws.rs.core.Feature;
 import javax.ws.rs.core.MediaType;
@@ -102,7 +102,7 @@
         final Reloader r = new Reloader();
 
         ResourceConfig rc = new ResourceConfig(classes);
-        UriConnegFilter.enableFor(rc, getMimeMappings(), null);
+        rc.property(ServerProperties.MEDIA_TYPE_MAPPINGS, getMimeMappings());
         rc.register(CsrfProtectionFilter.class);
 
 //        TODO - JERSEY2
Index: nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/RestCommandResourceProvider.java
===================================================================
--- nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/RestCommandResourceProvider.java	(revision 59996)
+++ nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/RestCommandResourceProvider.java	(working copy)
@@ -52,8 +52,7 @@
 import org.glassfish.admin.rest.readers.ParameterMapFormReader;
 import org.glassfish.admin.rest.resources.admin.CommandResource;
 import org.glassfish.hk2.api.ServiceLocator;
-import org.glassfish.jersey.filter.LoggingFilter;
-import org.glassfish.jersey.media.sse.OutboundEventWriter;
+import org.glassfish.jersey.media.sse.SseFeature;
 
 /**
  * Responsible for providing ReST resources for {@code asadmin} and {@code cadmin} communication.
@@ -62,7 +61,7 @@
  * @author sanjeeb.sahoo@oracle.com
  */
 public class RestCommandResourceProvider extends AbstractRestResourceProvider {
-    
+
     public RestCommandResourceProvider() {
         super();
     }
@@ -100,7 +99,7 @@
         r.add(MultipartFDPayloadReader.class);
         r.add(ParamsWithPayloadMultipartWriter.class);
         //SSE data
-        r.add(OutboundEventWriter.class);
+        r.add(SseFeature.class);
         r.add(AdminCommandStateJsonProvider.class);
         //ProgressStatus
         r.add(ProgressStatusJsonProvider.class);
@@ -109,7 +108,7 @@
 //        r.add(LoggingFilter.class);
         return r;
     }
-    
+
     @Override
     public Set<Class<?>> getResourceClasses(ServiceLocator habitat) {
         return getResourceClasses();
Index: nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/RestManagementResourceProvider.java
===================================================================
--- nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/RestManagementResourceProvider.java	(revision 59996)
+++ nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/adapter/RestManagementResourceProvider.java	(working copy)
@@ -53,6 +53,7 @@
 import org.glassfish.hk2.utilities.Binder;
 import org.glassfish.internal.api.ServerContext;
 import org.glassfish.jersey.jackson.JacksonFeature;
+import org.glassfish.jersey.media.sse.SseFeature;
 import org.glassfish.jersey.server.ResourceConfig;
 import org.jvnet.hk2.config.Dom;
 
@@ -164,7 +165,7 @@
         r.add(org.glassfish.admin.rest.provider.OptionsResultJsonProvider.class);
         r.add(org.glassfish.admin.rest.provider.OptionsResultXmlProvider.class);
 
-        r.add(org.glassfish.jersey.media.sse.OutboundEventWriter.class);
+        r.add(SseFeature.class);
         r.add(org.glassfish.admin.rest.provider.AdminCommandStateJsonProvider.class);
         r.add(org.glassfish.admin.rest.provider.ProgressStatusJsonProvider.class);
         r.add(org.glassfish.admin.rest.provider.ProgressStatusEventJsonProvider.class);
Index: nucleus/pom.xml
===================================================================
--- nucleus/pom.xml	(revision 59996)
+++ nucleus/pom.xml	(working copy)
@@ -114,7 +114,7 @@
         <gmbal.version>3.2.0-b003</gmbal.version>
         <woodstox.version>4.1.2</woodstox.version>
         <antlr.version>2.7.6</antlr.version>
-        <jersey.version>2.0-m12-1</jersey.version>
+        <jersey.version>2.0-SNAPSHOT</jersey.version>
         <jackson.version>1.9.8</jackson.version>
         <jettison.version>1.3.1</jettison.version>
         <guava.version>13.0</guava.version> 