Index: .
===================================================================
--- .	(revision 63571)
+++ .	(working copy)

Property changes on: .
___________________________________________________________________
Modified: svn:ignore
## -4,3 +4,4 ##
 target
 gfbuild.log
 .pom.xml.swp
+.DS_Store
Index: appserver
===================================================================
--- appserver	(revision 63571)
+++ appserver	(working copy)

Property changes on: appserver
___________________________________________________________________
Modified: svn:ignore
## -12,3 +12,4 ##
 .x3.swp
 .ql.errs.swp
 .idea
+.DS_Store
Index: appserver/web
===================================================================
--- appserver/web	(revision 63571)
+++ appserver/web	(working copy)

Property changes on: appserver/web
___________________________________________________________________
Modified: svn:ignore
## -5,3 +5,4 ##
 web (org.glassfish.web).iml
 web.iml
 web (org.glassfish.web).iml
+.DS_Store
Index: appserver/web/web-glue
===================================================================
--- appserver/web/web-glue	(revision 63571)
+++ appserver/web/web-glue	(working copy)

Property changes on: appserver/web/web-glue
___________________________________________________________________
Modified: svn:ignore
## -5,3 +5,4 ##
 webtier.iml
 webtier.iml
 .pom.xml.swp
+.DS_Store
Index: appserver/web/web-glue/src
===================================================================
--- appserver/web/web-glue/src	(revision 63571)
+++ appserver/web/web-glue/src	(working copy)

Property changes on: appserver/web/web-glue/src
___________________________________________________________________
Added: svn:ignore
## -0,0 +1 ##
+.DS_Store
Index: appserver/web/web-glue/src/main
===================================================================
--- appserver/web/web-glue/src/main	(revision 63571)
+++ appserver/web/web-glue/src/main	(working copy)

Property changes on: appserver/web/web-glue/src/main
___________________________________________________________________
Added: svn:ignore
## -0,0 +1 ##
+.DS_Store
Index: appserver/web/web-glue/src/main/java
===================================================================
--- appserver/web/web-glue/src/main/java	(revision 63571)
+++ appserver/web/web-glue/src/main/java	(working copy)

Property changes on: appserver/web/web-glue/src/main/java
___________________________________________________________________
Added: svn:ignore
## -0,0 +1 ##
+.DS_Store
Index: appserver/web/web-glue/src/main/java/org
===================================================================
--- appserver/web/web-glue/src/main/java/org	(revision 63571)
+++ appserver/web/web-glue/src/main/java/org	(working copy)

Property changes on: appserver/web/web-glue/src/main/java/org
___________________________________________________________________
Added: svn:ignore
## -0,0 +1 ##
+.DS_Store
Index: appserver/web/web-glue/src/main/java/org/glassfish
===================================================================
--- appserver/web/web-glue/src/main/java/org/glassfish	(revision 63571)
+++ appserver/web/web-glue/src/main/java/org/glassfish	(working copy)

Property changes on: appserver/web/web-glue/src/main/java/org/glassfish
___________________________________________________________________
Added: svn:ignore
## -0,0 +1 ##
+.DS_Store
Index: appserver/web/web-glue/src/main/java/org/glassfish/web
===================================================================
--- appserver/web/web-glue/src/main/java/org/glassfish/web	(revision 63571)
+++ appserver/web/web-glue/src/main/java/org/glassfish/web	(working copy)

Property changes on: appserver/web/web-glue/src/main/java/org/glassfish/web
___________________________________________________________________
Added: svn:ignore
## -0,0 +1 ##
+.DS_Store
Index: nucleus/pom.xml
===================================================================
--- nucleus/pom.xml	(revision 63572)
+++ nucleus/pom.xml	(working copy)
@@ -156,7 +156,7 @@
         <gmbal.version>4.0.0-b001</gmbal.version>
         <woodstox.version>4.1.2</woodstox.version>
         <antlr.version>2.7.6</antlr.version>
-        <jersey.version>2.10.2</jersey.version>
+        <jersey.version>2.12-SNAPSHOT</jersey.version>
         <jackson.version>2.3.2</jackson.version>
         <jettison.version>1.3.3</jettison.version>
         <jax-rs-api.version>2.0</jax-rs-api.version>
